cpp/src/arrow/python/arrow_to_pandas.cc

```diff
@@ -732,7 +745,7 @@ class ObjectBlock : public PandasBlock {
         }
       }
     } else if (type == Type::STRUCT) {
-      RETURN_NOT_OK(ConvertStruct(data, out_buffer));
+      RETURN_NOT_OK(ConvertStruct(options_, data, out_buffer));
```

Comment:
The `options_` parameter is only used in `ConvertStruct` and `ConvertListsLike`. Do we need it in all of the above (this code is probably due for some refactoring soon)?